pkg/planner/core/logical_plan_builder.go

```diff
@@ -784,8 +784,9 @@ func (b *PlanBuilder) coalesceCommonColumns(p *logicalop.LogicalJoin, leftPlan,
 		lNameMap := make(map[string]int, len(lNames))
 		rNameMap := make(map[string]int, len(rNames))
 		for _, name := range lNames {
-			// Natural join should ignore _tidb_rowid
-			if name.ColName.L == "_tidb_rowid" {
+			// Natural join should ignore _tidb_rowid and _tidb_commit_ts
+			if name.ColName.L == model.ExtraHandleName.L ||
+				name.ColName.L == model.ExtraCommitTSName.L {
```

Comment:
Do we need to wrap it as a function?

Reply:
I think 3 lines of code are not too much, and we are not repeating them everywhere. So I think we don't need to wrap a function for now.